lib/validator.js

```diff
@@ -236,7 +236,6 @@ function checkSubSchema(schema, parent, state) {
     }
     if (schema.not) {
         schema.not.should.be.an.Object();
-        schema.oneOf.should.not.be.an.Array();
```

Comment:
Because arrays are objects in JavaScript, I think this line should be retained, only with the typo `oneOf` changed to `not`.

Reply:
👍 Made this PR before seeing the comments on the commit. 

Looks like the `not.be.an.Object()` check works but with a less than ideal error message: 

`AssertionError: expected Array [ Object { not: Object { type: 'integer' } } ] not to be an array (false negative fail)`

The `not.be.an.Array()` check will never be reached. But its there just in case...